src/main/scala/io/github/pauljamescleary/petstore/validation/ValidationError.scala

```diff
@@ -4,5 +4,5 @@ import io.github.pauljamescleary.petstore.model.Pet
 
 sealed trait ValidationError extends Product with Serializable
 final case class PetAlreadyExistsError(pet: Pet) extends ValidationError
-final case object PetNotFoundError extends ValidationError
-final case object OrderNotFoundError extends ValidationError
+final case class PetNotFoundError() extends ValidationError
```

Comment:
Why the change from a `case object` over `case class`?

Reply:
The alternative is to define the return as `EitherT[F, PetNotFoundError.type, Pet]`

That isn't too bad, will post an update